plugins/playlist/src/components/PlaylistEditDialog/PlaylistEditDialog.tsx

```diff
@@ -74,13 +75,50 @@ export const PlaylistEditDialog = ({
 }: PlaylistEditDialogProps) => {
   const classes = useStyles();
   const identityApi = useApi(identityApiRef);
+  const playlistApi = useApi(playlistApiRef);
+  const playListApiData = playlistApi.getAllPlaylists({ editable: true });
+
+  const [editingOtherFields, setEditingOtherFields] = useState(false);
+
+  const fetchAndProcessData = async () => {
+    const playlistArray = await playListApiData;
+    return playlistArray;
+  };
 
   const { loading: loadingOwnership, value: ownershipRefs } =
     useAsync(async () => {
       const { ownershipEntityRefs } = await identityApi.getBackstageIdentity();
       return ownershipEntityRefs;
     }, []);
 
+  const nameIsUnique = async (
+    name: string,
+    isEditing: boolean,
+    originalName: string,
+  ) => {
+    if (!isEditing) {
+      const playlistArray = await fetchAndProcessData();
+      if (
+        playlistArray.some(
+          (playlistData: { name: string }) => playlistData.name === name,
+        )
+      ) {
+        return 'A playlist with this name already exists';
+      }
+    } else if (name !== originalName) {
+      const playlistArray = await fetchAndProcessData();
```

Comment:
Wait, is the code in this if statement and the one above exactly the same, or am I seeing it wrong?